pkgs/development/python-modules/onnx/default.nix

```
@@ -48,6 +48,8 @@ buildPythonPackage rec {
     substituteInPlace tools/protoc-gen-mypy.sh.in \
       --replace "/bin/bash" "${bash}/bin/bash"
 
+    substituteInPlace  requirements.txt \
+      --replace "protobuf >= 3.12.2, <= 3.20.1" ""
```

I think
```suggestion
      --replace "protobuf >= 3.12.2, < 4.0.0" ""
```
would be safer. In fact I see that that's what they've done upstream: https://github.com/onnx/onnx/pull/4535. (Ideally we would use the upstream patch using `fetchpatch` but I get some issues with the PyPI line endings when applying the patch.)

---

fixed!

---

I thought I fixed, but I could not build onnx (due to conflicting constraints). So I just use `pythonRelaxDepsHook` for both onnx and skl2onnx (as suggested https://github.com/NixOS/nixpkgs/pull/197194#issuecomment-1288060622)